port/linux/ipadapter.c

```diff
@@ -1493,6 +1593,20 @@ initialize_ip_context(ip_context_t *dev, size_t device,
   return true;
 }
 
+static void
+signal_event_thread(ip_context_t *dev)
+{
+  do {
+    if (write(dev->wakeup_pipe[1], "\n", 1) < 0) {
+      if (errno == EINTR) {
+        continue;
+      }
```

Comment:
handle ENOSPC - as return signal pipe is already set.

Reply:
Done.